compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -149,8 +149,8 @@ generateSrcPkgFromLf thisPkgId pkgMap pkg = do
             ]
         | LF.unModuleName (LF.moduleName m) == ["DA", "Internal", "Template"] =
             [ ""
-            , "class Template c where"
-            , "   signatory :: c -> [DA.Internal.LF.Party]"
+            , "class (Eq t, Show t) => Template t where"
+            , "   signatory :: t -> [DA.Internal.LF.Party]"
```

Comment:
@hurryabit Can you help me understand why the `Template` class is being replicated here missing most of the methods?